test/dynamo/test_graph_break_messages.py

```diff
@@ -498,6 +498,228 @@ def f(x):
             f(x)
         self.assertEqual(len(ws), 2)
 
+    def test_slice_with_tensor(self):
+        def fn(x, y):
+            return x[:y]
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(
+                torch.randn(10),
+                torch.tensor([3]),
+            ),
+            """\
+Dynamic slicing with Tensor arguments
+  Explanation: Creating slices with Tensor arguments is not supported. e.g. `l[:x]`, where `x` is a 1-element tensor.
+  Hint: It may be possible to write Dynamo tracing rules for this code. Please report an issue to PyTorch if you encounter this graph break often and it is causing performance issues.
+
+  Developer debug context: SliceVariable start: ConstantVariable(NoneType: None), stop: TensorVariable(), step: ConstantVariable(NoneType: None)
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return x[:y]""",
+        )
+
+    def test_observed_exception(self):
+        def fn():
+            raise RuntimeError("test")
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(),
+            """\
+Observed exception
+  Explanation: Dynamo found no exception handler at the top-level compiled function when encountering an exception. Exception will propagate outside the compiled region.
+  Hint: Dynamo has detected that tracing the code will result in an error when running in eager. Please double check that your code doesn't contain a similar error when actually running eager/uncompiled.
+  Hint: It may be possible to write Dynamo tracing rules for this code. Please report an issue to PyTorch if you encounter this graph break often and it is causing performance issues.
```

Comment:
What would we do in Dynamo? Raise the real exception?

Reply:
Yes, we could - I can update the last hint to reflect this more accurately.